ksa_compliance/jinja.py

```diff
@@ -19,19 +20,25 @@ def get_zatca_phase_1_qr_for_invoice(invoice_name: str) -> str:
 
 
 def get_qr_inputs(invoice_name: str) -> list:
-    sales_invoice = cast(SalesInvoice, frappe.get_doc("Sales Invoice", invoice_name))
-    seller_name = sales_invoice.company
+    invoice_doc = None
+    if frappe.db.exists("POS Invoice", invoice_name):
+        invoice_doc = cast(POSInvoice, frappe.get_doc("POS Invoice", invoice_name))
+    elif frappe.db.exists("Sales Invoice", invoice_name):
+        invoice_doc = cast(SalesInvoice, frappe.get_doc("Sales Invoice", invoice_name))
+    else:
+        return None
```

Comment:
The type annotation needs to be added to the `invoice_doc` definition in line 23. In this case, you should annotate it as being of type `Sales Invoice` since that's the common base type (`POS Invoice` inherits `Sales Invoice`)